<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>651</width>
    <height>627</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>651</width>
    <height>627</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>651</width>
    <height>627</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Кодування методом Шенона-Фано та Хафмана. Добинда Д.</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>156</x>
      <y>20</y>
      <width>341</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Monotype Corsiva</family>
      <pointsize>17</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Кодування методом Шенона-Фано</string>
    </property>
   </widget>
   <widget class="QPushButton" name="goCrypt">
    <property name="geometry">
     <rect>
      <x>539</x>
      <y>49</y>
      <width>91</width>
      <height>43</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Set</string>
    </property>
   </widget>
   <widget class="QPushButton" name="goDecrypt">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>150</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Courier New</family>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="text">
     <string>&lt;- Декодувати</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textToCrypt">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>521</width>
      <height>41</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
    <property name="placeholderText">
     <string>Default value</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textToDecrypt">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>100</y>
      <width>241</width>
      <height>101</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>310</y>
      <width>341</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Monotype Corsiva</family>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="text">
     <string>Оптимальний код методом Хафмана</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textToCryptH">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>390</y>
      <width>241</width>
      <height>101</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textToDectyptH">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>390</y>
      <width>241</width>
      <height>101</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="goCryptH">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>390</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Courier New</family>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Кодувати -&gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="goDecryptH">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>440</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Courier New</family>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>&lt;- Декодувати</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textToCryptCommon">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>100</y>
      <width>241</width>
      <height>101</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="goCrypt_2">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>100</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Courier New</family>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Кодувати -&gt;</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textToCryptHDefault">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>340</y>
      <width>521</width>
      <height>41</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>false</bool>
    </property>
    <property name="placeholderText">
     <string>Default value</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>539</x>
      <y>339</y>
      <width>91</width>
      <height>43</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Set</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>210</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>15</pointsize>
      <weight>75</weight>
      <italic>true</italic>
      <bold>true</bold>
      <strikeout>false</strikeout>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="text">
     <string>Коди:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>230</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>13</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>Ентропія = </string>
    </property>
   </widget>
   <widget class="QLabel" name="labelEnthropy">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>230</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>500</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>15</pointsize>
      <weight>75</weight>
      <italic>true</italic>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Коди:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>260</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>13</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>Середня довжина коду =</string>
    </property>
   </widget>
   <widget class="QLabel" name="everageH">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>260</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>560</y>
      <width>141</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>13</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>Середня довжина коду =</string>
    </property>
   </widget>
   <widget class="QLabel" name="everage">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>560</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>520</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>13</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>Ефективність коду =</string>
    </property>
   </widget>
   <widget class="QLabel" name="effective">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>230</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>230</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>13</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>Ефективність коду =</string>
    </property>
   </widget>
   <widget class="QLabel" name="effectiveH">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>520</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QTextEdit" name="outputLabel">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>230</y>
      <width>104</width>
      <height>81</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(240, 240, 240);</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="overwriteMode">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QTextEdit" name="outputHLabel">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>520</y>
      <width>104</width>
      <height>81</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(240, 240, 240);</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="overwriteMode">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>260</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>13</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>Біт = </string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>560</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>13</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>Біт = </string>
    </property>
   </widget>
   <widget class="QLabel" name="bit">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>260</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="bitH">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>560</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcdNumber">
    <property name="geometry">
     <rect>
      <x>605</x>
      <y>580</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value" stdset="0">
     <double>2016.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>520</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>13</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>Ентропія = </string>
    </property>
   </widget>
   <widget class="QLabel" name="labelEnthropyH">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>520</y>
      <width>47</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>651</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_File"/>
    <addaction name="actionCode_from_file"/>
    <addaction name="actionDecode_from_file"/>
    <addaction name="actionSave_all_to_file"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionABC"/>
    <addaction name="actionRandom"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
  </widget>
  <action name="actionOpen_File">
   <property name="text">
    <string>Open File</string>
   </property>
  </action>
  <action name="actionABC">
   <property name="text">
    <string>ABC</string>
   </property>
  </action>
  <action name="actionRandom">
   <property name="text">
    <string>Random</string>
   </property>
  </action>
  <action name="actionCode_from_file">
   <property name="text">
    <string>Code from file</string>
   </property>
  </action>
  <action name="actionDecode_from_file">
   <property name="text">
    <string>Decode from file</string>
   </property>
  </action>
  <action name="actionSave_all_to_file">
   <property name="text">
    <string>Save all to file</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
